### PR TITLE
eligibility checker

### PR DESCRIPTION
**Pull request by henryjiang-sfgov**
Runs a cron job or trigger manually to check eligibility and sends out an eligibility email to everyone in current tier.

How to run it locally:
curl -X GET -H 'ACCESS_KEY: 123456'  -H " application/json"  'http://127.0.0.1:8000/elgibility-checker?phase=phase1b'

### CODE REVIEW

File: service/resources/eligibility_checker.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                print("on_get error:")
+                print("{0}".format(err))
+                resp.status = falcon.HTTP_500
+                resp.body = json.dumps(jsend.error("{0}".format(err)))
+
+    # pylint: disable=no-member
+    def notifications(self, submission, phase):
+        """ sends notification about phase assignment """
+        # send confirmation email, updates confirmation_sent
+        timezone = pytz.timezone('America/Los_Angeles')
```
Comment (josh-chou) [inline service/resources/eligibility_checker.py:45]: can't we use LOCAL_TZ here?
or delete LOCAL_TZ if it's not being used anywhere.

File: service/resources/eligibility_checker.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        current_time = datetime.datetime.now(timezone)
+        status = NotificationService().send_notifications('eligiblilty', submission)
```
Comment (josh-chou) [inline service/resources/eligibility_checker.py:47]: spelling.  eligibility.
This string should get replaced with NotificationTypes.ELIGIBILITY after https://github.com/SFDigitalServices/vaccination-distribution/pull/19 is approved and merged.

File: service/microservice.py [hunk 3]
```diff
@@ -18,6 +19,7 @@ def start_service():
     api = falcon.API(middleware=[SQLAlchemySessionManager(create_session())])
     api.add_route('/welcome', Welcome())
     api.add_route('/submission', PhaseCalculator())
+    api.add_route('/elgibility-checker', EligibilityChecker())
```
Comment (josh-chou) [inline service/microservice.py:22]: spelling. eligibility-checker.

**Review by josh-chou — CHANGES_REQUESTED** [review]

File: service/resources/eligibility_checker.py (hunk 2 above)
Comment (henryjiang-sfgov) [reply service/resources/eligibility_checker.py:47]: yes I caught that.

**Review by josh-chou — APPROVED** [review]

File: service/resources/eligibility_checker.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    def on_get(self, _req, resp):
+        #pylint: disable=no-self-use
+        """
+            queries the database for eligibility send notifications
+        """
+        phase = _req.params['phase']
+        if phase:
+            try:
+                # pylint: disable=no-member
+                _records = get_eligible_list(self.session, phase) # querys the db for all individuals without phase assignment
```
Comment (cschlessinger) [inline service/resources/eligibility_checker.py:27]:
```suggestion
                _records = get_eligible_list(self.session, phase) # queries the DB for all individuals with the given phase assignment
```

File: service/resources/eligibility_checker.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            except Exception as err: # pylint: disable=broad-except
+                print("on_get error:")
+                print("{0}".format(err))
+                resp.status = falcon.HTTP_500
+                resp.body = json.dumps(jsend.error("{0}".format(err)))
+
+    # pylint: disable=no-member
+    def notifications(self, submission, phase):
+        """ sends notification about phase assignment """
+        # send confirmation email, updates confirmation_sent
```
Comment (cschlessinger) [inline service/resources/eligibility_checker.py:42]: This says confirmation email and confirmation_sent but it's passing an eligibility notification - if the comment is wrong, can you update it? Or if it is sending confirmation emails, what am I missing?

Comment (henryjiang-sfgov) [reply service/resources/eligibility_checker.py]: updated comment

File: service/resources/eligibility_checker.py (hunk 4 above)
Comment (henryjiang-sfgov) [reply service/resources/eligibility_checker.py]: updated comment

**Pull request merged**